core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -508,13 +517,40 @@ void updateFee() throws SepValidationException, AnchorException {
                     .receiveAmount(null)
                     .senderId(request.getSenderId())
                     .receiverId(request.getReceiverId())
-                    .clientId(token.getAccount())
+                    .clientId(getClientName())
                     .build())
             .getFee();
     infoF("Fee for request ({}) is ({})", request, fee);
     Context.get().setFee(fee);
   }
 
+  String getClientName() throws BadRequestException {
+    return getClientName(Context.get().getSep10Jwt().getAccount());
+  }
+
+  String getClientName(String account) throws BadRequestException {
+    ClientConfig client = clientsConfig.getClientConfigBySigningKey(account);
+    if (!sep10Config.getClientAttributionAllowList().isEmpty()) {
+      // check if the client is not in the allow list, client should be denied
+      if (client != null
+          && !sep10Config.getClientAttributionAllowList().contains(client.getName())) {
+        client = null;
+      }
+    }
+
+    if (sep10Config.isClientAttributionRequired()) {
+      if (client == null) {
+        throw new BadRequestException("Client not found");
+      }
+    } else {
+      // client attribution not required
+      if (client == null) {
+        return null;
+      }
+    }
+    return client.getName();
+  }
+
```

Comment:
The second if check is no longer necessary since we put validation in PropertySep10Config to make sure `getClientAttributionAllowList` (renamed to `getAllowedClientDomains`) is never empty and always returns the intersection of `clientAllowList` and clientsConfig.clilents list